### PR TITLE
rename `type` to `asgi.event.type` in ASGI instrumentation

### LINKED ISSUE #2055: ASGI instrumentation creates a 'type' attribute
See https://github.com/open-telemetry/opentelemetry-python-contrib/blob/7166de673fa9dd7abc178bec3103d4adebb05e0d/instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py#L681

This results in an attribute like `{'type': 'http.request.body'}`, which seems wrong. `type` is a _super_ generic key and I don't think leaking the details of the web framework spec (`http.request.body` is an ASGI specific string) is very helpful to users.

Can we get rid of this?

### PR DESCRIPTION
**Pull request by samuelcolvin**
# Description

As per https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2055#issuecomment-1918148101, `type` in `opentelemetry-instrumentation-asgi` is not a good name for something specific to asgi.

As per discussion on https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2145#issuecomment-1933035603, instead of removing `type`, we should change its name.

This replaces #2145.

Fixes #2055

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] `opentelemetry-instrumentation-asgi` tests updated

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [x] Documentation has been updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Fixed
```
Comment (lzchen) [inline CHANGELOG.md:10]: We probably want to start indicating breaking changes as more users start to use our instrumentations so please rename this to `### Breaking Changes`.

**Review by lzchen — APPROVED** [review]
LGTM. @ocelotl  @aabmass  @srikanthccv  

Is everyone okay with the name `asgi.event.type`?

**Comment by srikanthccv** [comment]
do individual SIGs have flexibility in choosing the attribute names that don't apply to other frameworks/languages? In the past, we removed all attributes except the ones spec'd by sem conventions.

**Comment by samuelcolvin** [comment]
I've resolved conflicts again, would be great if this could be merged.

> do individual SIGs have flexibility in choosing the attribute names that don't apply to other frameworks/languages? In the past, we removed all attributes except the ones spec'd by sem conventions.

I understand that, and I originally argued to remove this attribute, but since it exists and is extremely helpful, I don't think it really makes sense to remove it at this stage.

**Comment by lzchen** [comment]
> We probably want to start indicating breaking changes as more users start to use our instrumentations so please rename this to ### Breaking Changes.

**Review by srikanthccv — APPROVED** [review]

**Comment by srikanthccv** [comment]
@lzchen Updated the changelog

**Review by lzchen — APPROVED** [review]

**Pull request merged**